components/supervisor/pkg/terminal/service.go

```diff
@@ -48,15 +58,22 @@ func (srv *MuxTerminalService) RegisterREST(mux *runtime.ServeMux, grpcEndpoint
 	return api.RegisterTerminalServiceHandlerFromEndpoint(context.Background(), mux, grpcEndpoint, []grpc.DialOption{grpc.WithInsecure()})
 }
 
-// Open opens a new terminal running the login shell
+// Open opens a new terminal running the shell
 func (srv *MuxTerminalService) Open(ctx context.Context, req *api.OpenTerminalRequest) (*api.OpenTerminalResponse, error) {
-	cmd := exec.Command(srv.LoginShell[0], srv.LoginShell[1:]...)
+	return srv.OpenWithOptions(ctx, req, TermOptions{
+		ReadTimeout: 5 * time.Second,
+	})
+}
+
+// OpenWithOptions opens a new terminal running the shell with given options
+func (srv *MuxTerminalService) OpenWithOptions(ctx context.Context, req *api.OpenTerminalRequest, options TermOptions) (*api.OpenTerminalResponse, error) {
+	cmd := exec.Command(srv.DefaultShell, "-i")
```

Comment:
This makes the assumption that all shells anyone would want to use as `SHELL` support `-i`.
I couldn't find/google prior conventions for the SHELL variable, i.e. if others would include the `-i` in the SHELL command, or add it like we do here.

Reply:
I will drop `-i`